docs/models/advanced/modeldata.md

```diff
@@ -0,0 +1,61 @@
+Model Data
+==========
+
+Model data is the successor of "extended blockstates" allowing modders to pass in extra data to be used in [`IBakedModel::getQuads`][getquads] when rendering blocks. This is particularly useful in complex models such as for camouflage blocks which use other block's models as an example. This is not to be confused with a [`TileEntityRenderer`][ter] which is used for dynamic rendering such as for animated blocks like a chest or any blocks which cannot be defined by a list of `BakedQuad`s.
+
+IBakedModels vs TERs
+------------------
+
+An [`IBakedModel`][bakedmodel] simply provides a list of `BakedQuad`s to the `BlockModelRenderer` which are then drawn to the screen. There are other properties available which define aspects which effect rendering such as whether the model supports ambient occlusion (a rendering effect). Most [`IBakedModel`][bakedmodel]s come from when a model file is loaded and then "baked" so that the model does not need to be loaded again.
+
+A [`TileEntityRenderer`][ter] or [`TER`][ter] (previously [`TileEntitySpecialRenderer`][ter] or [`TESR`][ter]) is used to render blocks in a way that cannot be represented with a static baked model (JSON, OBJ, B3D, others).
+
+A [`TER`][ter] has direct access to the `TileEntity` that the renderer is used for whereas an [`IBakedModel`][bakedmodel] only has access to the `BlockState` the model represents, the `Direction` which represents the side which is being rendered (or null), an instance of `Random` for randomised textures and the `IModelData` parameter. The [`IBakedModel::getQuads`][getquads] is unaware of where the block is in the world hence where the `IModelData` parameter steps in to provide extra information from the `TileEntity` using `TileEntity::getModelData`, and from the world using `IBakedModel::getModelData`.
+
+Unlike a [`TER`][ter] which renders every frame, an [`IBakedModel`][bakedmodel] only provides the quads for drawing upon request (which occurs when `World::markBlockRangeForRenderUpdate` is called).
+
+Model Data flow
+------------------------
+
+Model data is first created in the `TileEntity::getModelData` method which by default returns an `EmptyModelData` instance meaning that there is no data provided. This is then passed to the [`IBakedModel::getModelData`][getmodeldata] method as the `tileData` parameter. By default, the [`IBakedModel::getModelData`][getmodeldata] returns the `tileData` parameter. The result of [`IBakedModel::getModelData`][getmodeldata] is then the value of the `modelData` parameter in the [`IBakedModel::getQuads`][getquads] function.
```

Comment:
I think the convention is ClassName#MethodName for non-static. (instead of :: )

Reply:
I've changed that now thanks :)